### ISSUE TITLE
Precompiled NIFs for freebsd

### ISSUE THREAD

**Issue by maltekrupa** [elixir-nx/explorer#617]
Hi,

I'm using livebook on a freebsd host. When I tried to `Mix.install` explorer, the following error appeared:

```
== Compilation error in file lib/explorer/polars_backend/native.ex ==
** (RuntimeError) precompiled NIF is not available for this target: "amd64-portbld-freebsd13.1".
The available targets are:
 - aarch64-apple-darwin
 - aarch64-unknown-linux-gnu
 - aarch64-unknown-linux-musl
 - riscv64gc-unknown-linux-gnu
 - x86_64-apple-darwin
 - x86_64-pc-windows-msvc
 - x86_64-pc-windows-gnu
 - x86_64-unknown-linux-gnu
 - x86_64-unknown-linux-musl
    lib/explorer/polars_backend/native.ex:11: (module)
```

To reproduce this issue, start livebook or iex on a freebsd host and run the following:
```
Mix.install([{:explorer, "~> 0.5.0"}])
```

I tried my luck by [adding a freebsd target to the list of jobs](https://github.com/maltekrupa/explorer/pull/1/files) to see if it would at least build it but I ran into an [error that is too far out of my comfort zone](https://github.com/maltekrupa/explorer/actions/runs/5277102837/jobs/9544685707?pr=1).

The error states: `cannot find -lmemstat` and I guess it means the what it says, but then there is this issue https://github.com/cross-rs/cross/pull/1166 that states that they added memstat to the freebsd images ... and now I don't know how to proceed.

To sum it up: Is there any chance to have precompiled NIFs for freebsd? Thanks.

**Comment by josevalim** [comment]
If someone contributes a PR to the build system, we will be glad to precompile it, but we don’t have the knowledge to tackle it ourselves.

**Comment by maltekrupa** [comment]
Makes sense. I'll see if I can get any further. Thank you.
